### PR TITLE
feat: add profile access_model, repr and missing bindings

### PR DESCRIPTION
**Pull request by vishwa2710**
## Release Notes

- **New Features**
  - Introduced `access_model` method in the `Profile` class, allowing users to access the profile model.
  - Added new methods in the `Model` class for type checking and casting: `is_tabulated`, `is_transform`, `as_tabulated`, and `as_transform`.
  - Enhanced string representation for `Tabulated` and `Transform` classes.
  - Added a new test suite for the `Transform` class, covering various functionalities.

- **Bug Fixes**
  - Improved error handling in the `Profile` class to ensure safe access to the model.

- **Tests**
  - Added tests for the `access_model` method to verify functionality and error handling.
  - Expanded test coverage for the `Transform` class with multiple new test cases.

- **Documentation**
  - Updated method documentation for clarity on new functionalities.

### CODE REVIEW

File: bindings/python/test/flight/test_profile.py [hunk 1]
```diff
@@ -134,6 +134,18 @@ def test_profile_target(self, alignment_target: Profile.Target):
         assert alignment_target is not None
         assert isinstance(alignment_target, Profile.Target)
 
+    def test_access_model(self, profile: Profile):
+        model = profile.access_model()
+
+        assert model is not None
+
```
Comment (vishwa2710) [inline bindings/python/test/flight/test_profile.py:141]:
```suggestion
        assert model is not None
```

File: bindings/python/test/flight/test_profile.py [hunk 2]
```diff
@@ -134,6 +134,18 @@ def test_profile_target(self, alignment_target: Profile.Target):
...
+
```
Comment (vishwa2710) [inline bindings/python/test/flight/test_profile.py:142]:
```suggestion
        assert model is not None

```

File: include/OpenSpaceToolkit/Astrodynamics/Flight/Profile.hpp [hunk 3]
```diff
@@ -208,6 +208,16 @@ class Profile
     /// @return True if profile is defined
     bool isDefined() const;
 
+    /// @brief Access profile model
+    ///
+    /// @code{.cpp}
+    ///              Profile profile = { ... };
+    ///              const Unique<Model>& model = profile.accessModel();
```
Comment (vishwa2710) [inline include/OpenSpaceToolkit/Astrodynamics/Flight/Profile.hpp:215]:
```suggestion
    ///              const Model& model = profile.accessModel();
```

File: include/OpenSpaceToolkit/Astrodynamics/Flight/Profile.hpp [hunk 4]
```diff
@@ -208,6 +208,16 @@ class Profile
...
     bool isDefined() const;
 
+    /// @brief Access profile model
+    ///
+    /// @code{.cpp}
+    ///              Profile profile = { ... };
+    ///              const Model& model = profile.accessModel();
+    /// @endcode
+    ///
+    /// @return Reference to the profile model
```
Comment (apaletta3) [inline include/OpenSpaceToolkit/Astrodynamics/Flight/Profile.hpp:218]:
```suggestion
    /// @return Const reference to the profile model
```

File: test/OpenSpaceToolkit/Astrodynamics/Flight/Profile.test.cpp [hunk 5]
```diff
@@ -370,6 +381,7 @@ TEST_F(OpenSpaceToolkit_Astrodynamics_Flight_Profile, InertialPointing)
     }
 }
 
+// TBI: Should move these tests to the Transform test suite and only test interface here
```
Comment (apaletta3) [inline test/OpenSpaceToolkit/Astrodynamics/Flight/Profile.test.cpp:384]: Can you make a GH issue and ticket for this? Or open another MR to do this?

File: test/OpenSpaceToolkit/Astrodynamics/Flight/Profile/Transform.test.cpp [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+            anInstant,
+            -position_GCRF.accessCoordinates(),
+            -velocity_GCRF.accessCoordinates(),
+            Quaternion::Unit(),
+            Vector3d(0.0, 0.0, 0.0)  // TBM: Artificially set to 0 for now.
+        );
+    };
+
+    const Transform transform = Transform(DynamicProvider(dynamicProviderGenerator), Frame::GCRF());
+    EXPECT_TRUE(transform.isDefined());
```
Comment (apaletta3) [inline test/OpenSpaceToolkit/Astrodynamics/Flight/Profile/Transform.test.cpp:105]:
```suggestion
    EXPECT_NO_THROW(Transform transform_ (DynamicProvider(dynamicProviderGenerator), Frame::GCRF()));
```

File: test/OpenSpaceToolkit/Astrodynamics/Flight/Profile/Transform.test.cpp [hunk 7]
```diff
@@ -0,0 +1,215 @@
...
+TEST_F(OpenSpaceToolkit_Astrodynamics_Flight_Profile_Models_Transform, CalculateStateAt)
+{
+    {
+        const Instant instant = Instant::DateTime(DateTime(2024, 1, 29, 0, 0, 15), Scale::UTC);
+
+        const State state = transform_.calculateStateAt(instant);
+
+        EXPECT_EQ(state.getInstant(), instant);
+
+        EXPECT_TRUE(state.accessFrame() == Frame::GCRF());
```
[the remark this answers is not in the document]
Comment (apaletta3) [reply test/OpenSpaceToolkit/Astrodynamics/Flight/Profile/Transform.test.cpp:145]: Agree with this

File: test/OpenSpaceToolkit/Astrodynamics/Flight/Profile/Transform.test.cpp [hunk 8]
```diff
@@ -0,0 +1,218 @@
...
+        EXPECT_TRUE(Transform::InertialPointing(orbit_, Quaternion::Unit()).isDefined());
+    }
+}
+
+TEST_F(OpenSpaceToolkit_Astrodynamics_Flight_Profile_Models_Transform, NadirPointing)
+{
+    {
+        EXPECT_TRUE(Transform::NadirPointing(orbit_, Orbit::FrameType::VVLH).isDefined());
+    }
+}
```
Comment (apaletta3) [inline test/OpenSpaceToolkit/Astrodynamics/Flight/Profile/Transform.test.cpp:218]: I think you should assert a couple more things here, like the type of transform or something to verify that you actually are inertially pointing / nadir pointing with this transform

**Review by apaletta3 — APPROVED** [review]
LGTM. Couple small things to address

File: test/OpenSpaceToolkit/Astrodynamics/Flight/Profile/Transform.test.cpp (hunk 8 above)
Comment (vishwa2710) [reply test/OpenSpaceToolkit/Astrodynamics/Flight/Profile/Transform.test.cpp:217]: Those are asserted in the Profile tests. Eventually we can move those tests from Profile to here.

File: test/OpenSpaceToolkit/Astrodynamics/Flight/Profile.test.cpp (hunk 5 above)
Comment (vishwa2710) [reply test/OpenSpaceToolkit/Astrodynamics/Flight/Profile.test.cpp:384]: Will open a follow up MR when I have some time. Don't think it's worth making a GH issue.

**Pull request merged**